### DISCUSSION TITLE
How to see benchmarks when running from source

### DISCUSSION THREAD

**Discussion by jokorn** [ghostfolio/ghostfolio#1142] [Q&A] [answered]
Hi

You recently added benchmarks to https://ghostfol.io/home/market. However, when I run from source I only see the headers and no data are shown:
![Screenshot 2022-08-12 21 30 53](https://user-images.githubusercontent.com/7759201/184444647-5fa1f6ad-6e1d-42c0-bf6e-ab245e9c10c3.png)
Are any manual changes needed to make this work, like we have to do for the Fear & Greed Index (https://github.com/ghostfolio/ghostfolio/discussions/833)?

**Comment by dtslvr** [comment, marked as answer]
Hi @jokorn

Yes, you need to create the property `BENCHMARKS` in the table `Property` to make it work. For example:

|key|value|
|---|------|
|`BENCHMARKS`|`[{"dataSource":"YAHOO","symbol":"VOO"}]`

Make sure that `YAHOO`:`VOO` is also present in the table `SymbolProfile`.

**Reply by jokorn** [reply]
Hi dtslvr

That worked great. 
I added the symbol profile for S&P 500 index by running this .ts file with `npx ts-node` (I cannot find out how to add via `npx prisma studio` as a lot of the fields are undefined):

```
import {PrismaClient} from '@prisma/client';

const prisma = new PrismaClient();

async function main() {
  await prisma.symbolProfile.create({
    data: {
    assetClass: 'EQUITY',
    currency: 'USD',
    dataSource: 'YAHOO',
    name: 'S&P 500',
    symbol: '^GSPC'
  },
  });
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
});
```
I then added the `BENCHMARKS` property in the `Property` table with `value` `[{"dataSource":"YAHOO","symbol":"^GSPC"}]` using `npx prisma studio`.

After gathering data for S&P 500 using the `Admin Control`, it now works great! Before gathering data via Admin Control, the Market tab displayed an error message in the bottom of the screen and the terminal showed this error:

```
Error: [big.js] Invalid number
    at parse (/home/username/ghostfolio/node_modules/big.js/big.js:142:13)
    at new Big (/home/username/ghostfolio/node_modules/big.js/big.js:109:9)
    at Big.P.div (/home/username/ghostfolio/node_modules/big.js/big.js:346:16)
    at /home/username/ghostfolio/dist/apps/api/webpack:/ghostfolio/apps/api/src/app/benchmark/benchmark.service.ts:52:10
    at Array.map (<anonymous>)
    at BenchmarkService.<anonymous> (/home/username/ghostfolio/dist/apps/api/webpack:/ghostfolio/apps/api/src/app/benchmark/benchmark.service.ts:48:31)
    at Generator.next (<anonymous>)
    at fulfilled (/home/username/ghostfolio/node_modules/tslib/tslib.js:111:62)
```

Thanks a lot!

**Reply by dtslvr** [reply]
> Error: [big.js] Invalid number

Fixed via #1177.

**Comment by jokorn** [comment]
Hi

After #1248, how do I add benchmarks when running from source?

I tried changing the value of `BENCHMARKS` in `Property` to `[{"symbolProfileId":"c29dec38-de04-44a9-94e0-ee9116cf1559"}]` where "c29dec38-de04-44a9-94e0-ee9116cf1559" is an `id` from the `symbolProfile` table, but it does not seem to work.

I receive no errors in the terminal or console, but the benchmarks are not showing.

Can you help again?

**Reply by dtslvr** [reply]
Hi @jokorn

The new format is correct like this:

```
[{"symbolProfileId":"c29dec38-de04-44a9-94e0-ee9116cf1559"}]
```

The result of the benchmarks is cached in _Redis_ for 4 hours, probably that's why it is still empty.

**Reply by jokorn** [reply]
Perfect. Thanks a lot. It appeared after 4 hours.

**Reply by dtslvr** [reply]
Nice 😊
